src/libsyntax/parse/parser/expr.rs

```diff
@@ -186,6 +186,12 @@ impl<'a> Parser<'a> {
                 self.last_type_ascription = None;
                 return Ok(lhs);
             }
+            (true, Some(AssocOp::Greater)) if self.restrictions.contains(
```

Comment:
I can't say I'm following the changes in this particular file and how it is hanging together & will need some help with that. As this is subtle, it would be good if @petrochenkov could also give this a look.